### DISCUSSION TITLE
Link items with lines/arrows

### DISCUSSION THREAD

**Discussion by TeChn4K** [samuelarbibe/dnd-timeline#47] [Ideas]
Hello,

Thanks a lot for this great lib! An headless Gant/Timeline lib was really missing in the JS ecosystem.

Do you have any "arrow/link between items" feature on the roadmap? And if yes, how complex would such a feature be?

**Comment by samuelarbibe** [comment]
Hey!  
I was actually thinking about this a few days ago.  
I don't think it will be an integral part of the library, but I will implement an example of gantt-style dependency arrow's.  
When I finish, I'll be sure to update you ❤️

**Reply by step2341** [reply]
Hi samuel,

Great work so far. In addition to dependencies, are we able to restrict an item to move within a row?

**Reply by samuelarbibe** [reply]
Hey step, can you be more specific?  
Do you want to limit an item to be draggable only to specific rows? do you want items in certain rows to be disabled?

**Reply by samuelarbibe** [reply]
@TeChn4K BTW, haven't forgotten about you!    
I am playing around with implementing a connection functionality, without using any external libraries, and minimal added code.  
<img width="1368" alt="Screenshot 2024-07-29 at 16 57 14" src="https://github.com/user-attachments/assets/b3ebe43d-6aad-47fa-b8d1-190bafae24e7">  
I am not entirely happy with this implementation, will update when I find something better.

**Reply by TeChn4K** [reply]
Pleased to hear it @samuelarbibe !
It will be a nice feature addition for sure

**Reply by TeChn4K** [reply]
BTW, how do you handle these connection lines/arrows? pure CSS? SVG? Canvas?

**Reply by samuelarbibe** [reply]
I use only SVG for drawing the arrows. I simply look for the source and target items' positions, and use a function to create an svg path. This is to avoid adding extra renders.

**Reply by step2341** [reply]
Hi @samuelarbibe,

With a ghantt type configuration, in the sidebar there would need to be items/rows, sometimes with the ability for subrows and make them collapsible/expandable. often ghantts use a table structure in the sidebar so that a row can have multiple cells. 

But then since each row represents a task/item the dnd behavior in the timeline is then locked to that rows horizontal axis. ie the task/item can grow in duration, and start/end, but is not able to be moved to a different row, as that would be mean it is now a different task.

<img width="883" alt="image" src="https://github.com/user-attachments/assets/f4fdb95c-6e19-47c9-ad5d-b2863baf5bb4">

Using dnd-kits use sortable hook in the table would be good though as that would allow reordering of rows.

**Reply by samuelarbibe** [reply]
This is an interesting configuration, it can probably be done using tanstack's `react-table` library.  
I have recently joined a new team at work, so I have a very limited amount of time to perform R&D on external examples.  
Unfortunatly, I prefer using my free time doing R&D on the library's core features and performance, So I won't be able to help with the development of this.  

That being said, feel free to contact me for consultations.   
Have fun ❤️

**Reply by vitalygashkov** [reply]
> I am playing around with implementing a connection functionality, without using any external libraries, and minimal added code

Thanks for the lightweight and flexible library :)

Is there any news on implementation of arrows between items? I suddenly need this functionality for a Gantt chart in a project of the company I work in, so I would be glad if you could share a draft or any hints in this direction.

**Comment by samuelarbibe** [comment]
Hey! an update on the subject.  
I have been playing around with implementing an example without using external libraries.  
So far, I have been able to create a minimal example, but I have encountered a situation where the arrows don't show up correctly on mount, but only after performing some change on the timeline and causing a re-render (like moving an item).  
  
You can checkout the branch and the changes here: #61   
  
I will continue to try and work on it.
